### ISSUE TITLE
fine-tuning results

### ISSUE THREAD

**Issue by Matjaz12** [HeverLaw/sam3.1-training#1]
Thanks for sharing the code! I’m curious -- have you experimented with fine-tuning the tracker on a VOS dataset? If so, would you be willing to share the hyperparameters you used, along with a training curve or any insights from the process (e.g., training stability, required batch size, and approximate training time)?

**Comment by HeverLaw** [comment]
I trained the model with our surgical VOS dataset, and the final model is working better than SAM2. Here is the dataset https://arxiv.org/abs/2511.16618 we used, although we didn't test on the general VOS dataset.

**Comment by HeverLaw** [comment]
This is the train all loss in our dataset. 
<img width="370" height="263" alt="Image" src="https://github.com/user-attachments/assets/b9e63aa0-fb72-4e54-be13-bcb079664f8b" />

**Comment by Matjaz12** [comment]
Hey, thanks! Interesting. What is the difference between the bold blue curve and the translucent blue curve? I’m guessing one is a smoothed version of the other. In that case, your results are consistent with mine -- I have my own reimplementation of the SAM3 tracker and have found that the focal loss can be quite noisy, while the other losses, including (i) Dice loss, (ii) object-prediction cross-entropy, and (iii) IoU-prediction MAE, are much smoother and decrease nicely.

<img width="400" height="200" alt="Image" src="https://github.com/user-attachments/assets/f07fe706-7ede-4d6d-baae-0c4d39c91335" />

<img width="400" height="200" alt="Image" src="https://github.com/user-attachments/assets/aad21981-c19f-41ed-a1a0-8e338715896f" />

(Modified version of SAM3 trained on a mixture of YTVOS, SA-V, and MOSEv2)

**Comment by HeverLaw** [comment]
These are two identical experiments. And I also have a similar cross-entropy, which aligns with SAM2 training.
